aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPlugin.java

```diff
@@ -72,61 +85,179 @@ public void disable() {
     @Override
     public void recordEvent(@NonNull String eventName) {
 
+        final com.amazonaws.mobileconnectors.pinpoint.analytics.AnalyticsEvent pinpointEvent =
+                analyticsClient.createEvent(eventName);
+        analyticsClient.recordEvent(pinpointEvent);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void recordEvent(@NonNull AnalyticsEvent analyticsEvent) {
+    public void recordEvent(@NonNull AnalyticsEvent analyticsEvent)
+            throws AnalyticsException {
 
+        final com.amazonaws.mobileconnectors.pinpoint.analytics.AnalyticsEvent pinpointEvent =
+                analyticsClient.createEvent(analyticsEvent.getName());
+
+        if (analyticsEvent.getProperties() != null) {
+            for (Map.Entry<String, Property<?>> entry : analyticsEvent.getProperties().get().entrySet()) {
+                if (entry.getValue() instanceof StringProperty) {
+                    pinpointEvent.addAttribute(entry.getKey(), ((StringProperty) entry.getValue()).getValue());
+                } else if (entry.getValue() instanceof DoubleProperty) {
+                    pinpointEvent.addMetric(entry.getKey(), ((DoubleProperty) entry.getValue()).getValue());
+                } else {
+                    throw new AnalyticsException("Invalid property type detected.",
+                            "AmazonPinpointAnalyticsPlugin support only StringProperty or DoubleProperty." +
+                                    "Refer tio the docuemntation for details.");
+                }
+            }
+        }
+        analyticsClient.recordEvent(pinpointEvent);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void registerGlobalProperties(Map<String, Object> properties) {
-
+    public void registerGlobalProperties(@NonNull Properties properties) {
+        throw new UnsupportedOperationException("This operation has not been implemented yet.");
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void unregisterGlobalProperties(Set<String> keys) {
-
+    public void unregisterGlobalProperties(@NonNull Set<String> keys) {
+        throw new UnsupportedOperationException("This operation has not been implemented yet.");
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void flushEvents() {
-
+        analyticsClient.submitEvents();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public String getPluginKey() {
-        return null;
+        return "amazonPinpointAnalyticsPlugin";
```

Comment:
I believe this should be "awsPinpointAnalyticsPlugin" as that is the name of the key that is added to amplifyconfiguration.json by `amplify add analytics` command.  This causes an NPE in `configure` because it's trying to fetch the configuration using the wrong key name.